org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/TestVMType.java

```diff
@@ -74,7 +80,19 @@ public IStatus validateInstallLocation(File installLocation) {
 	 */
 	@Override
 	public File detectInstallLocation() {
-		return new File("testresources");
+		return getInstallLocation();
+	}
+
+	protected static File getInstallLocation() {
+		Bundle bundle = Platform.getBundle(JavaLanguageServerTestPlugin.PLUGIN_ID);
+		try {
+			URL url = FileLocator.toFileURL(bundle.getEntry("/testresources"));
```

Comment:
beware, [`FileLocator.toFileURL` doesn't escape spaces](https://bugs.eclipse.org/bugs/show_bug.cgi?id=145096), which causes some issues down the line.